python/plugins/processing/algs/qgis/Intersection.py

```diff
@@ -96,8 +96,10 @@ def processAlgorithm(self, progress):
                     int_geom = QgsGeometry(geom.intersection(tmpGeom))
                     if int_geom.wkbType() == QgsWkbTypes.Unknown or QgsWkbTypes.flatType(int_geom.geometry().wkbType()) == QgsWkbTypes.GeometryCollection:
                         int_com = geom.combine(tmpGeom)
-                        int_sym = geom.symDifference(tmpGeom)
-                        int_geom = QgsGeometry(int_com.difference(int_sym))
+                        int_geom = QgsGeometry()
+                        if int_com is not None:
```

Comment:
In QGIS 3 you can't test geometries this way. You just test it direct, ie:

    if int_com: